From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: dasd: log sense for fatal errors.
References: bnc#445100

Symptom:     When a fatal error occures no sense data is printed.
Problem:     During Hyper PAV work the log sense call was accidentally 
             removed.
Solution:    Call dasd_log_sense() for fatal errors.

Acked-by: John Jolly <jjolly@suse.de>

---
 drivers/s390/block/dasd.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: linux-sles11/drivers/s390/block/dasd.c
===================================================================
--- linux-sles11.orig/drivers/s390/block/dasd.c
+++ linux-sles11/drivers/s390/block/dasd.c
@@ -1773,6 +1773,11 @@ restart:
 			goto restart;
 		}
 
+		/* log sense for fatal error */
+		if (cqr->status == DASD_CQR_FAILED) {
+			dasd_log_sense(cqr, &cqr->irb);
+		}
+
 		/* First of all call extended error reporting. */
 		if (dasd_eer_enabled(base) &&
 		    cqr->status == DASD_CQR_FAILED) {
